cuda_core/cuda/core/experimental/_program.py

```diff
@@ -6,6 +6,157 @@
+        if self.device_stack_protector is not None:
+            self.formatted_options.append(f"--device-stack-protector={'true' if self.device_stack_protector else 'false'}".encode())
+        if self.define_macro is not None:
+            if isinstance(self.define_macro, tuple):
+                self.formatted_options.append(f"--define-macro={self.define_macro[0]}={self.define_macro[1]}".encode())
+            else:
+                self.formatted_options.append(f"--define-macro={self.define_macro}".encode())
+        if self.undefine_macro is not None:
+            self.formatted_options.append(f"--undefine-macro={self.undefine_macro}".encode())
+        if self.include_path is not None:
+            self.formatted_options.append(f"--include-path={self.include_path}".encode())
+        if self.pre_include is not None:
+            self.formatted_options.append(f"--pre-include={self.pre_include}".encode())
+        if self.no_source_include is not None:
+            self.formatted_options.append("--no-source-include".encode())
+        if self.std is not None:
+            self.formatted_options.append(f"--std={self.std}".encode())
+        if self.builtin_move_forward is not None:
+            self.formatted_options.append(f"--builtin-move-forward={'true' if self.builtin_move_forward else 'false'}".encode())
+        if self.builtin_initializer_list is not None:
+            self.formatted_options.append(f"--builtin-initializer-list={'true' if self.builtin_initializer_list else 'false'}".encode())
+        if self.disable_warnings is not None:
+            self.formatted_options.append("--disable-warnings".encode())
+        if self.restrict is not None:
+            self.formatted_options.append("--restrict".encode())
+        if self.device_as_default_execution_space is not None:
+            self.formatted_options.append("--device-as-default-execution-space".encode())
+        if self.device_int128 is not None:
+            self.formatted_options.append("--device-int128".encode())
+        if self.optimization_info is not None:
+            self.formatted_options.append(f"--optimization-info={self.optimization_info}".encode())
+        if self.display_error_number is not None:
+            self.formatted_options.append("--display-error-number".encode())
+        if self.no_display_error_number is not None:
+            self.formatted_options.append("--no-display-error-number".encode())
+        if self.diag_error is not None:
+            self.formatted_options.append(f"--diag-error={self.diag_error}".encode())
+        if self.diag_suppress is not None:
+            self.formatted_options.append(f"--diag-suppress={self.diag_suppress}".encode())
+        if self.diag_warn is not None:
+            self.formatted_options.append(f"--diag-warn={self.diag_warn}".encode())
+        if self.brief_diagnostics is not None:
+            self.formatted_options.append(f"--brief-diagnostics={'true' if self.brief_diagnostics else 'false'}".encode())
```

Comment:
The ternary statement seems like a pattern that might benefit from a helper function.